### PR TITLE
Improve extract speed

### PR DESCRIPTION
**Pull request by vvmruder**
Improve municipality handling, this might break existing adapted non standard sources!!!

This is related to #823

### CODE REVIEW

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/standard/sources/municipality.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4629388&bid=15517973)

**Comment by peterschaer** [comment]
Thanks! This is waaaay faster than before (on a small real estate from 3secs to 1secs for a JSON extract!). So it looks good to me.

**Comment by vvmruder** [comment]
@peterschaer Nice that it works. Do you know if you use some adapted elements in your project which you need to change? Or was it working out of the box?

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/standard/sources/municipality.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4629388&bid=15517973)

**Comment by vvmruder** [comment]
And in addition: Did you tried the capabilities request? The fix is interfering with this. So I would be interested if output is still like you would expect it.

**Comment by peterschaer** [comment]
It was working out of the box, I didn't have to change anything. however the capabilities request has different output. before it had 346 municipalities (the same number as in the municipality table), now it has 702 municipalities. there are a lot of duplicates.

**Comment by vvmruder** [comment]
Ups... I will have a look at it.

**Comment by vvmruder** [comment]
@peterschaer Could you please try again? I hope my fix solve this issue.

**Comment by vvmruder** [comment]
And maybe can you try several times to see if the duplicates are increasing if the error still exists?

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pyramid_oereb/standard/sources/municipality.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4629388&bid=15517973)

**Comment by peterschaer** [comment]
very good, the duplicates are gone even if i try multiple times.

**Comment by vvmruder** [comment]
Juhu. I will wait for review of the others. But in my opinion we can merge this. It is quite an improvement. Even if it only really bothers cantons with really many municipalities :smiley:

**Comment by vvmruder** [comment]
@peterschaer Thanks for catching this in the code!

**Comment by vvmruder** [comment]
@kdeininger I'am glad you see it like this. I think this too. I will wait for comment from @jwkaltz and then we can merge.

File: pyramid_oereb/lib/readers/municipality.py [hunk 1]
```diff
@@ -33,10 +33,13 @@ def read(self):
             :ref:`api-pyramid_oereb-lib-records-municipality-municipalityrecord`. Otherwise the API like way
             the server works would be broken.
 
+        Args:
+            fosnr (int or None): The federal number of the municipality defined by the statistics office.
+
         Returns:
-            list of pyramid_oereb.lib.records.municipality.MunicipalityRecord:
+            pyramid_oereb.lib.records.municipality.MunicipalityRecord:
             The list of all found records. Since these are not filtered by any criteria the list simply
```
[the remark this answers is not in the document]
Comment (kdeininger) [reply pyramid_oereb/lib/readers/municipality.py]: Yes, it still is a list. But this is a good point. The comment should be adapted anyway as the list may be filtered now.

**Pull request merged**